File: androidApp/src/main/kotlin/zinqq/app/screens/settings/AdvancedScreen.kt
package zinqq.app.screens.settings

import androidx.compose.foundation.background
import androidx.compose.foundation.clickable
import androidx.compose.foundation.layout.Column
import androidx.compose.foundation.layout.fillMaxWidth
import androidx.compose.foundation.layout.padding
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.runtime.LaunchedEffect
import androidx.compose.runtime.collectAsState
import androidx.compose.runtime.getValue
import androidx.compose.runtime.mutableStateOf
import androidx.compose.runtime.remember
import androidx.compose.runtime.setValue
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.platform.LocalClipboardManager
import androidx.compose.ui.text.AnnotatedString
import androidx.compose.ui.text.font.FontFamily
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import kotlinx.coroutines.delay
import zinqq.app.R
import zinqq.app.WalletHolder
import zinqq.app.nav.Route
import zinqq.app.theme.ZinqqTheme

/**
 * The PWA's Advanced (U17, R12; `Advanced.tsx`): the Node ID copy card with
 * the 2,000 ms "Copied!" flash, then the Balance and Peers rows. The node id
 * is cached on [zinqq.app.UiState] (queried per refresh — `node_id()` needs
 * a running node); like the PWA's not-ready gate, the card simply doesn't
 * render before the first successful start.
 */
@Composable
fun AdvancedScreen(
    holder: WalletHolder,
    onBack: (() -> Unit)?,
    onOpenRow: (Route) -> Unit,
) {
    val state by holder.state.collectAsState()
    val colors = ZinqqTheme.colors
    val clipboard = LocalClipboardManager.current
    var copied by remember { mutableStateOf(false) }

    // The PWA's 2,000 ms copied flash (Advanced.tsx:56-62).
    LaunchedEffect(copied) {
        if (copied) {
            delay(2_000)
            copied = false
        }
    }

    SettingsScaffold(title = "Advanced", onBack = onBack) {
        Column(
            modifier = Modifier
                .fillMaxWidth()
                .verticalScroll(rememberScrollState())
                .padding(16.dp),
        ) {
            state.nodeId?.let { nodeId ->
                Column(
                    modifier = Modifier
                        .fillMaxWidth()
                        .padding(bottom = 16.dp)
                        .clip(RoundedCornerShape(12.dp))
                        .background(colors.darkElevated)
                        .clickable {
                            clipboard.setText(AnnotatedString(nodeId))
                            copied = true
                        }
                        .padding(16.dp),
                ) {
                    Text(
                        text = "Node ID",
                        color = colors.onDarkMuted,
                        fontSize = 12.sp,
                        fontWeight = FontWeight.Medium,
                    )
                    Text(
                        text = nodeId,
                        color = colors.onDark,
                        fontFamily = FontFamily.Monospace,
                        fontSize = 12.sp,
                        lineHeight = 18.sp,
                        modifier = Modifier.padding(top = 4.dp),
                    )
                    Text(
                        text = if (copied) "Copied!" else "Tap to copy",
                        color = colors.onDarkMuted,
                        fontSize = 12.sp,
                        modifier = Modifier.padding(top = 8.dp),
                    )
                }
            }
            ADVANCED_ROWS.forEach { row ->
                SettingsRowItem(
                    row = row,
                    iconRes = if (row.label == "Balance") {
                        R.drawable.ic_card_plus
                    } else {
                        R.drawable.ic_users
                    },
                    onClick = row.destination?.let { { onOpenRow(it) } },
                )
            }
        }
    }
}

File: iosApp/iosApp/Screens/Settings/AdvancedScreen.swift
import SwiftUI
import UIKit

/// The PWA's Advanced (U22, R12; `Advanced.tsx`): the Node ID copy card with
/// the 2,000 ms "Copied!" flash, then the Balance and Peers rows. The node id
/// is cached on `WalletModel` (fetched on refresh until cached — `nodeId()`
/// needs a running node); like the PWA's not-ready gate, the card simply
/// doesn't render before the first successful start. Mirrors Android's
/// `AdvancedScreen`.
struct AdvancedScreen: View {
    @ObservedObject var model: WalletModel
    var onBack: (() -> Void)?
    let onOpenRow: (Route) -> Void

    @Environment(\.zinqqColors) private var colors
    @State private var copied = false

    var body: some View {
        SettingsScaffold(title: "Advanced", onBack: onBack) {
            ScrollView {
                VStack(spacing: 0) {
                    if let nodeId = model.nodeId {
                        nodeIdCard(nodeId)
                            .padding(.bottom, 16)
                    }
                    ForEach(advancedRows, id: \.label) { row in
                        SettingsRowItem(
                            row: row,
                            systemImage: row.label == "Balance" ? "creditcard" : "person.2",
                            onClick: row.destination.map { destination in
                                { onOpenRow(destination) }
                            }
                        )
                    }
                }
                .padding(16)
            }
        }
        // The PWA's 2,000 ms copied flash (Advanced.tsx:56-62).
        .autoReset($copied, afterMs: 2_000)
    }

    private func nodeIdCard(_ nodeId: String) -> some View {
        Button {
            UIPasteboard.general.string = nodeId
            copied = true
        } label: {
            VStack(alignment: .leading, spacing: 0) {
                Text("Node ID")
                    .font(ZinqqFont.sans(12, weight: .medium))
                    .foregroundColor(colors.onDarkMuted)
                Text(nodeId)
                    .font(.system(size: 12, design: .monospaced))
                    .foregroundColor(colors.onDark)
                    .multilineTextAlignment(.leading)
                    .padding(.top, 4)
                Text(copied ? "Copied!" : "Tap to copy")
                    .font(ZinqqFont.sans(12))
                    .foregroundColor(colors.onDarkMuted)
                    .padding(.top, 8)
            }
            .frame(maxWidth: .infinity, alignment: .leading)
            .padding(16)
            .background(colors.darkElevated)
            .clipShape(RoundedRectangle(cornerRadius: 12))
        }
        .accessibilityLabel(copied ? "Node ID copied" : "Copy node ID")
    }
}
